examples/data-objects/inventory-app/src/inventoryListTrio.ts

```diff
@@ -0,0 +1,92 @@
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+
+import { IInventoryListUntyped } from "./interfaces";
+import { sharedTreeInventoryListFactory } from "./sharedTreeInventoryList";
+import { legacySharedTreeInventoryListFactory } from "./legacySharedTreeInventoryList";
+
+const legacySharedTreeInventoryListKey = "legacySharedTreeInventoryList";
+const sharedTreeInventoryListKey = "sharedTreeInventoryList";
+const sharedTreeForHookKey = "sharedTreeForHook";
+
+/**
+ * A DataObject providing three approaches to inventory lists - both LegacySharedTree and SharedTree
+ * contained within DataObjects, as well as SharedTree used directly by the view in a custom React hook.
+ */
+export class InventoryListTrio extends DataObject {
+	private _legacySharedTreeInventoryList: IInventoryListUntyped | undefined;
+	private _sharedTreeInventoryList: IInventoryListUntyped | undefined;
+	private _sharedTreeForHook: ISharedTree | undefined;
+
+	public get legacySharedTreeInventoryList() {
+		if (this._legacySharedTreeInventoryList === undefined) {
+			throw new Error("Not initialized properly");
+		}
+		return this._legacySharedTreeInventoryList;
+	}
+
+	public get sharedTreeInventoryList() {
+		if (this._sharedTreeInventoryList === undefined) {
+			throw new Error("Not initialized properly");
+		}
+		return this._sharedTreeInventoryList;
+	}
+
+	public get sharedTreeForHook(): ISharedTree {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		return this._sharedTreeForHook!;
+	}
+
+	protected async initializingFirstTime() {
+		const legacySharedTreeInventoryList =
+			await legacySharedTreeInventoryListFactory.createChildInstance(this.context);
+
+		const sharedTreeInventoryList = await sharedTreeInventoryListFactory.createChildInstance(
+			this.context,
+		);
+
+		const sharedTreeForHook = this.runtime.createChannel(
+			undefined,
+			new SharedTreeFactory().type,
+		) as ISharedTree;
+
+		this.root.set(legacySharedTreeInventoryListKey, legacySharedTreeInventoryList.handle);
+		this.root.set(sharedTreeInventoryListKey, sharedTreeInventoryList.handle);
+		this.root.set(sharedTreeForHookKey, sharedTreeForHook.handle);
+	}
+
+	protected async hasInitialized() {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		this._legacySharedTreeInventoryList = await this.root
+			.get<IFluidHandle<IInventoryListUntyped>>(legacySharedTreeInventoryListKey)!
+			.get();
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		this._sharedTreeInventoryList = await this.root
+			.get<IFluidHandle<IInventoryListUntyped>>(sharedTreeInventoryListKey)!
+			.get();
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		this._sharedTreeForHook = await this.root
+			.get<IFluidHandle<ISharedTree>>(sharedTreeForHookKey)!
+			.get();
+	}
+}
+
+// REV: One interesting challenge is that SharedTree and LegacySharedTree have the same Type ("SharedTree")
```

Comment:
good catch. We should just use a different "type" field for tree2 since we already have docs in the wild with "SharedTree" in the attributes blob. @taylorsw04 @CraigMacomber let's chat about this at standup on Monday (hopefully one of us will remember to bring it up :))